vortex-flatbuffers/src/owned.rs

```diff
@@ -0,0 +1,140 @@
+use vortex_error::VortexResult;
+
+use crate::FlatBuffer;
+
+pub trait Owned {
+    fn try_new(buffer: FlatBuffer) -> VortexResult<Self>
+    where
+        Self: Sized,
+    {
+        Self::try_new_with_loc(buffer, 0)
+    }
+
+    fn try_new_with_loc(buffer: FlatBuffer, loc: usize) -> VortexResult<Self>
+    where
+        Self: Sized;
+}
+
+macro_rules! make_owned {
```

Comment:
this is a stripped-down macro inspired by https://docs.rs/flatbuffers-owned/latest/flatbuffers_owned/